### PR TITLE
Fix RMW_Connext_Client::is_service_available for micro

### LINKED ISSUE #128: Services are broken if using connext micro
**System Info**

- _OS_
  - Ubuntu 22.04
- *ROS version and installation type*
  - humble/binary  (using humble branch because of #127)
- *RTI Connext DDS version and installation type*
  - Micro 3.0.3.28/rtipkg
- *RMW version or commit hash*
  - 94bde7556d069bee8f9540171a9b87593c3be51e

**Bug Description**

ROS2 service clients are unable to detect any available (matching) service instance.

**Expected Behavior**

ROS2 service clients should be able to detect available (matching) service instances.

**How to Reproduce**

1. setup a workspace with the following layout:
  - `rmw_connext_ws`
    - `rti_connext_dds-6.1.2` (containing the whole Pro package as well as the micro package in `rti_connext_dds_micro-3.0.3.28`)
    - `src/rmw_connextdds`
2. put `COLCON_IGNORE` in rmw_connext_ws/rti_connext_dds-6.1.2
3. run `docker run -it -v $PWD:/home -w /home osrf/ros:humble-desktop in rmw_connext_ws`
4. run `source ./rti_connext_dds-6.1.2/resource/scripts/rtisetenv_x64Linux4gcc7.3.0.bash`
5. run `colcon build`
6. run `source install/setup.bash`
7. run `export RMW_IMPLEMENTATION=rmw_connextddsmicro`
8. run `export RMW_CONNEXT_INITIAL_PEERS="_shmem://, 127.0.0.1"`
9. run `ros2 run examples_rclcpp_minimal_service service_main & ros2 run examples_rclcpp_minimal_client client_main`

This will result in this error:
grafik

**Workarounds**

The issues stems (at least partially) from the fact that RMW_Connext_Client::is_service_available uses
- DDS_DataWriter_get_matched_subscriptions
- DDS_DataWriter_get_matched_publications 

which are both only implemented as stubs in connext micro: both will usually return DDS_RETCODE_UNSUPPORTED.

Inspired by https://github.com/ros2/rmw_connext/blob/5557440c0a16dd272812587f17801975a7fa7632/rmw_connext_cpp/src/rmw_service_server_is_available.cpp#L93 I could come up with https://github.com/cwecht/rmw_connextdds/commit/969171a1cdb75ad0e633800b8f4e6ad7dbae31eb .

With that fix there is no error anymore but the service is found.

Please note that I also head to adjust the RMW_CONNEXT_LIMIT_DEFAULT_MAX to 256 (32 and 64 didn't work).

### PR DESCRIPTION
**Pull request by cwecht**
Fixed #128.

### CODE REVIEW

**Review by clalancette — CHANGES_REQUESTED** [review]
Can you please target this at the `rolling` branch first?  We can backport it from there.

**Comment by cwecht** [comment]
rebase done.

**Review by clalancette — CHANGES_REQUESTED** [review]
One minor nit to fix, then I think this will be good to run CI on.

File: rmw_connextdds_common/src/common/rmw_impl.cpp [hunk 1]
```diff
@@ -2576,6 +2576,11 @@ RMW_Connext_Client::enable()
 rmw_ret_t
 RMW_Connext_Client::is_service_available(bool & available)
 {
+#if RMW_CONNEXT_DDS_API == RMW_CONNEXT_DDS_API_MICRO
+  available =  0 < this->request_pub->subscriptions_count() && 0 < this->reply_sub->publications_count();
```
[the remark this answers is not in the document]
Comment (cwecht) [reply rmw_connextdds_common/src/common/rmw_impl.cpp]: Applied suggested fix and applied ament_uncrustify.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19303/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13834/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20023/)

**Pull request merged**